lib/CGameState.cpp

```diff
@@ -3300,9 +3300,6 @@ void CPathfinder::initializeGraph()
 
 void CPathfinder::calculatePaths()
 {
-	assert(hero);
-	assert(hero == getHero(hero->id));
-
 	bool flying = hero->hasBonusOfType(Bonus::FLYING_MOVEMENT);
```

Comment:
What is "legacy" about it? This is teh simplest and recommended methof for bonus checking, used everywhere in mechanics.